### ISSUE TITLE
[FEAT] Rootless mode

### ISSUE THREAD

**Issue by Mithrandir2k18** [linuxserver/docker-openssh-server#79]
### Is this a new feature request?

- [X] I have searched the existing issues

### Wanted change

Possibility to have the server run as a non-root user within the container.

### Reason for change

I need to run this in a k8s cluster and we have security policies that by default don't allow pods to run as root. Since this binds to a port that doesn't need to run as root anyway (2222), it should be possible to just run sshd without any permissions at all.

**Comment by aptalca** [comment]
> Possibility to have the server run as a non-root user within the container.

It does not run as root. It runs as the PUID/PGID set

**Comment by thespad** [comment]
The container will run rootless but will not run as a non-root user; the latter is not a scenario we can currently support.

**Comment by Mithrandir2k18** [comment]
@thespad it just came up again in our organisation, could you please add some details/reasons as to why that cannot be supported currently?

**Comment by thespad** [comment]
TL;DR

sshd doesn't run as root but the container needs to due to limitations of the supervisor we use. Note that true rootless operation, where the container runs as "root" with user namespace remapping, *should* work.

There are plans in the pipeline to support limited non-root container usage but they're dependent on upstream changes outside of our control.

**Comment by Mithrandir2k18** [comment]
Thank you for the quick response. If this is something you are waiting for, should this feature request remain open with some kind of postponed label? Is there maybe an upstream github issue we could mention here to track any changes? I realize this is low priority because sshd already can operate without root, but having this container run fully rootless would be very nice to have.

**Comment by thespad** [comment]
It's not something specific to this image so there's no need to track it here. I also don't have an ETA and can't say 100% that it'll work for any given image until we're in a position to implement and test the changes.
